clients/drcachesim/tools/func_view.cpp

```diff
@@ -224,18 +240,22 @@ func_view_t::process_memref(const memref_t &memref)
         return true;
     switch (memref.marker.marker_type) {
     case TRACE_MARKER_TYPE_FUNC_RETADDR: {
-        assert(shard->last_func_id != -1);
-        const auto &info = id2info_[shard->last_func_id];
+        const auto &info = get_info_for_last_func_id(shard);
         bool was_nested = shard->nesting_level > 0;
-        if (shard->prev_noret)
-            --shard->nesting_level;
-        std::string indent(shard->nesting_level * 4, ' ');
+        if (shard->prev_noret) {
+            if (was_nested) {
+                --shard->nesting_level;
+            } else {
+                std::cerr << "WARNING: Unnested FUNC_RETADDR\n";
```

Comment:
I had to go read the code to understand this: so prev_noret is for the id *before* the one we just read; so not the id associated with this RETADDR.  Please change this warning string to be more clear: maybe "WARNING: last function was noret, yet nesting was somehow unrolled prior to the new function" or something like that should help future readers/users understand it.

Reply:
I have to admit I don't really understand the markers well enough for this code to make full sense, but I made the message more descriptive.